haskell-src/Concordium/Types/Transactions.hs

```diff
@@ -139,6 +141,60 @@ instance S.Serialize TransactionSignature where
                     accumulateSigs (Map.insert idx sigmap accum) (Just idx) (count - 1)
         TransactionSignature <$> accumulateSigs Map.empty Nothing len
 
+-----------------------------------------------------------------
+
+-- * JSON representation of a signed/partially-signed transaction
+
+-----------------------------------------------------------------
+
+-- | A 'SignedTransaction' is a transaction that is signed by an account (the signer)
+-- with some keys. The representation might be a fully signed transaction ready to be
+-- sent on-chain or a partially-signed transaction that needs additional signatures
+-- added to be ready to be sent on-chain.
+--
+-- The `ToJSON` instance has the purpose converting the object into a human-readable
+-- representation ready to be printed into a JSON file. This file can be shared among
+-- different tools of the Concordium ecosystem for adding additional signatures.
+--
+--  The chosen representation is the minimal necessary data needed to construct the
+-- 'TransactionSignHash' which is the value that is signed by the signer. The
+-- 'TransactionSignHash' and 'payloadSize' should be re-computed when processing a
+-- 'SignedTransaction' (e.g. when adding signatures or sending the transaction on-chain).
+--
+-- The representation has a `version` field.
+data SignedTransaction = SignedTransaction
+    { -- | A version to distinguish between future formats of signed/partially-signed transactions.
+      -- The initial version is 1 and will be incremented for every new format.
+      stVersion :: !Int,
```

Comment:
This is an implementation detail of the JSON encoding, so it shouldn't actually be included in the data structure. I recommend defining the `ToJSON` and `FromJSON` instances explicitly, and including the version handling as part of that.